src/hardware/memory.cpp

```diff
@@ -472,17 +472,21 @@ bool mem_unalignedreadd_checked(PhysPt address, Bit32u * val) {
 	return false;
 }
 
-bool mem_unalignedwritew_checked(PhysPt address,Bit16u val) {
-	if (mem_writeb_checked(address,(Bit8u)(val & 0xff))) return true;val>>=8;
-	if (mem_writeb_checked(address+1,(Bit8u)(val & 0xff))) return true;
```

Comment:
The warning is right; I skimmed right past the val>>=8 over there on the right.
Logic and readability is normal now.

Reply:
Yeah, me too. And lack of whitespace made it easy to miss 'return' in there as well.